right/src/config_parser/parse_config.c

```diff
@@ -11,41 +11,22 @@
 #include "config.h"
 #include "mouse_controller.h"
 #include "mouse_keys.h"
+#include "layer.h"
+#include <stdint.h>
+#include "parse_module_config.h"
+#include "layer_switcher.h"
+#include "usb_report_updater.h"
+#include "debug.h"
 
     uint16_t DataModelMajorVersion = 0;
     uint16_t DataModelMinorVersion = 0;
     uint16_t DataModelPatchVersion = 0;
 
     bool PerKeyRgbPresent = false;
 
-static parser_error_t parseModuleConfiguration(config_buffer_t *buffer)
+void readRgbColor(config_buffer_t *buffer, rgb_t* keyActionColors, key_action_color_t keyActionColor)
 {
-    uint8_t id = ReadUInt8(buffer);
-    uint8_t pointerMode = ReadUInt8(buffer);  // move vs scroll
```

Comment:
@kareltucek If I read correctly this PR then it drops the old `ModuleConfiguration` logic without any backward compatibility. Agent had not used any data from the old `ModuleConfiguration`. Agent just serialised and deserialised the firmware data.
The default user configuration of the Agent had not contain any default value of `ModuleConfiguration` so if someone reseted the user configuration then it deleted the `ModuleConfiguration`.
Its also mean the factory default does not contains any `ModuleConfiguration`.

Based on the above facts I think I could also drop the old `ModuleConfiguration` code and write the new logic and I don't have to care about the backward compatibility.

Reply:
I just now talked with Laci and he confirmed I can drop the old code